Mathlib/CategoryTheory/SmallObject/TransfiniteCompositionLifting.lean

```diff
@@ -0,0 +1,168 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.SmallObject.WellOrderInductionData
+import Mathlib.CategoryTheory.MorphismProperty.TransfiniteComposition
+import Mathlib.CategoryTheory.LiftingProperties.Basic
+
+/-!
+# The left lifting property is stable under transfinite composition
+
+Let `C` be a category, and `p : X ⟶ Y` be a morphism in `C`. In this file,
+we show that a transfinite composition of morphisms that have the left
+lifting property with respect to `p` also has the left lifting property with
+respect to `p`, see `HasLiftingProperty.transfiniteComposition.hasLiftingProperty_ι_app_bot`.
+
+TODO: Given `P : MorphismProperty C`, deduce that the class of morphisms
+that have the left lifting property with respect to `P` is stable
+by transfinite composition.
+
+-/
+
+universe w v u
+
+namespace CategoryTheory
+
+open Category Limits
+
+variable {C : Type u} [Category.{v} C]
+
+namespace HasLiftingProperty
+
+variable {W : MorphismProperty C}
+  {J : Type w} [LinearOrder J] [OrderBot J]
+
+namespace transfiniteComposition
+
+variable {F : J ⥤ C} (c : Cocone F) (hc : IsColimit c)
+  {X Y : C} (p : X ⟶ Y)
+  (f : F.obj ⊥ ⟶ X) (g : c.pt ⟶ Y)
+
+/-- Given a colimit cocone `c` for a functor `F : J ⥤ C` from a well-ordered type,
+and maps `p : X ⟶ Y`, `f : F.obj ⊥ ⟶ X`, `g : c.pt ⟶ Y`, this structure
+contains the data of a map `F.obj j ⟶ X` such that `F.map (homOfLE bot_le) ≫ f' = f`
+and `f' ≫ p = c.ι.app j ≫ g`. -/
+@[ext]
+structure SqStruct (j : J) where
```

Comment:
Could you please motivate the name, maybe by drawing an ascii-art diagram in the docstring?

Reply:
Thanks for the suggestion! I have added a comment about the general strategy of the proof and some asciiart.